proto/contract.proto

```diff
@@ -111,16 +114,20 @@ message Egress {
   // destination is the sink where events are sent.
   string destination = 2;
 
+  //destination CA Cert is the CA Cert used for HTTPS communication through destination
+  string destinationCACerts = 15;
+
   oneof replyStrategy {
-    // Send the response to an url
-    string replyUrl = 3;
+    // Sends message to reply url with the CA Cert provided
+    replyUrlAndCert reply = 3;
```

Comment:
Isn't this a breaking change? 

Like is it possible to serialize and deserialize a proto message of an earlier version with this new version? Ideally, it should be possible to serialize and deserialize a proto message of this new version with the old version (in case of downgrades).
Basically, we need to make sure that the new proto schema is forward and backward compatible with the previous version.

If the above are not possible, what I'd is to add a separate field on the `Egress`: `string replyCACerts = 15;` and we need to make sure we set that field when `ReplyStrategy` is `replyUrl`

Reply:
@pierDipi Can we deprecate reply url and add replyUrlAndCert?

Reply:
We can, does that ensure backward and forward compatibility?

We need to think about:
- what happens (or how to solve) when a previous version of the dispatcher sees a message with `replyUrlAndCert` (it doesn't know anything about `replyUrlAndCert`)
- what happens (or how to solve) when the new version sees the old `replyUrl` and there isn't a `replyUrlAndCert` (this might happen when the upgrade is in progress and so the controller is at the previous version and dispatcher is at the "current version")

In theory, to solve those by adding and deprecating we need to set both fields for a few releases but that clashes with "oneof"